scheduler/system_sched_test.go

```diff
@@ -1307,6 +1307,80 @@ func TestSystemSched_Queued_With_Constraints(t *testing.T) {
 	if val, ok := h.Evals[0].QueuedAllocations["web"]; !ok || val != 0 {
 		t.Fatalf("bad queued allocations: %#v", h.Evals[0].QueuedAllocations)
 	}
+
+}
+
+// No errors reported when constraints prevent placement
+func TestSystemSched_ConstraintErrors(t *testing.T) {
+	h := NewHarness(t)
+
+	var node *structs.Node
+	// Register some nodes
+	// the tag "aaaaaa" is hashed so that the nodes are processed
+	// in an order other than good, good, bad
+	for _, tag := range []string{"aaaaaa", "foo", "foo", "foo"} {
+		node = mock.Node()
+		node.Meta["tag"] = tag
+		node.ComputeClass()
+		require.Nil(t, h.State.UpsertNode(h.NextIndex(), node))
+	}
+
+	// Mark the last node as ineligible
+	node.SchedulingEligibility = structs.NodeSchedulingIneligible
+	// node.ComputeClass() // should only need to be updated at registration?
+
+	// Make a job with a partially matching constraint
```

Comment:
Suggested rewording for this - make a job with a constraint that matches a subset of nodes